### ISSUE TITLE
[bug] useWeb3Modal 4.1.1

### ISSUE THREAD

**Issue by nsnull0** [WalletConnect/web3modal#2059]
### Link to minimal reproducible example

web3Modal 4.1.1 is buggy

### Summary

Hello team,

So basically i just update to latest web3modal
```
"@web3modal/wagmi": "^4.1.1",
"viem": "^2.8.13",
"wagmi": "^2.5.11"
```
I found out that `await open({ view: 'Networks' })` doesn't want to switch to selected network, it just navigate to connect wallet again, the page need to be refresh in order to make it work.

if I am using legacy version
```
npm install @web3modal/wagmi wagmi@1.4.13 viem@1.21.4 --legacy-peer-deps
```
it is working fine

fyi, no code changes apart from different method and setup when integrate with latest version.

Is there anything I am missing? Is there away to switch to selected network without refresh?

## evidence:
**4.1.1**
https://github.com/WalletConnect/web3modal/assets/26643922/39980e21-cbf1-4e44-929d-39cd2afa3c30

**3.5.7**
https://github.com/WalletConnect/web3modal/assets/26643922/969d7ce0-69bb-4a25-995b-a02fc494ea91

### List of related npm package versions

"@web3modal/wagmi": "^4.1.1",
"viem": "^2.8.13",
"wagmi": "^2.5.11"

**Comment by glitch-txs** [comment]
Are you able to reproduce the issue here: https://lab.web3modal.com/
Could you share steps to reproduce it if so? Thanks!

**Comment by nsnull0** [comment]
> ### Link to minimal reproducible example
> web3Modal 4.1.1 is buggy
> 
> ### Summary
> Hello team,
> 
> So basically i just update to latest web3modal
> 
> ```
> "@web3modal/wagmi": "^4.1.1",
> "viem": "^2.8.13",
> "wagmi": "^2.5.11"
> ```
> 
> I found out that `await open({ view: 'Networks' })` doesn't want to switch to selected network, it just navigate to connect wallet again, the page need to be refresh in order to make it work.
> 
> if I am using legacy version
> 
> ```
> npm install @web3modal/wagmi wagmi@1.4.13 viem@1.21.4 --legacy-peer-deps
> ```
> 
> it is working fine
> 
> fyi, no code changes apart from different method and setup when integrate with latest version.
> 
> Is there anything I am missing? Is there away to switch to selected network without refresh?
> 
> ## evidence:
> **4.1.1** https://github.com/WalletConnect/web3modal/assets/26643922/39980e21-cbf1-4e44-929d-39cd2afa3c30
> 
> **3.5.7** https://github.com/WalletConnect/web3modal/assets/26643922/969d7ce0-69bb-4a25-995b-a02fc494ea91
> 
> ### List of related npm package versions
> "@web3modal/wagmi": "^4.1.1", "viem": "^2.8.13", "wagmi": "^2.5.11"



https://github.com/WalletConnect/web3modal/assets/26643922/ea16f928-ea13-4b72-809e-cb71cf970e21

I tried on https://lab.web3modal.com/ 
seems I could reproduce the issue, almost similiar issue like on my evidence.
-
btw, now i am reverting back to 3.5.7.. until this issue fixed.
currently my dapps can't even connect to any injected wallet eventho.

**Comment by nsnull0** [comment]
@glitch-txs btw, I am following the tutorial at doc for email wallet.

**Comment by nsnull0** [comment]
well yes and also no.
 please check the whole video...
also the evidence..
IT should successfully ask permission to connect again.. but in this case is not, it just showing error connection instead of asking again to injected wallet in this case metamask.. the flag `isConnected` still true, but it somehow losing connection...

**Comment by nsnull0** [comment]
@glitch-txs could you check again please.. thank you

**Comment by nsnull0** [comment]
the thing is,, 
3.5.7 is asking permission to reconnect.
4.1.1 just showing error connection, IF user is not advanced web3 user, they may not understand what to do, also why `isConnected` still true eventho it disconnected like you said.

**Comment by glitch-txs** [comment]
@nsnull0 could you try again, this should be fixed by now

**Comment by nsnull0** [comment]
Thank you for follow up the issue.. I guess it is from version `4.1.6`... let me test on `4.1.6` soon..

**Comment by nsnull0** [comment]
https://github.com/WalletConnect/web3modal/assets/26643922/c341139f-de5d-4eba-9d64-9773c3351184

seems it still there. it can be triggered while connecting to metamask.

this is the code for connect, i am using useWeb3Modal()
```
try {
            await open()
        } catch (error) {
            
        }
```

this is the code for switch button
```
try {
            await open({view: "Networks"})
        } catch (error) {
            
        }
```
also i add 1 button for debugging which is `<w3m-network-button/>` also same effect.

also I print out the `isConnected` from `const { isConnected, address, chain } = useAccount()`
```
<p>
                            isConnected Debug: {isConnected ? "true" : "false"}
                        </p>
```

this is the code for disconnecting
```
if (isConnected) {
            try {
                await disconnectAsync()
            } catch (error) {
                window.alert(error)
            }

            return
        }
```

my dependencies package:
"@web3modal/wagmi": "4.1.6",
"viem": "2.7.19",
"wagmi": "2.5.7"

**Comment by glitch-txs** [comment]
Could you create a small repo with a minimal implementation example to reproduce the issue? And share the steps to reproduce it?

**Comment by nsnull0** [comment]
sure, i'll share here later..

**Comment by nsnull0** [comment]
@glitch-txs I think i found out my problem while creating fresh new project.

### 2 issues resolved:
1. I was accidentally using `Web3ModalProvider` at many different place, it caused overwritten thus the account information is gone, it need to be refreshed manually.
2. tried again with 4.1.1 , confirmed it still got disconnection bug, which already fixed with 4.1.6

now it fixed with the latest version, working as expected seamlessly.

Thank you for the support!

**Comment by nsnull0** [comment]
@glitch-txs  sorry it seems the disconnection bug still lingering with 4.1.6 version 
here is the minimum project to reproduce the bug
`https://github.com/nsnull0/minweb3modal`

### Steps to reproduce 100%
- connect wallet
- refresh page manually
wagmi will show status connected but web3modal is disconnected.

ref videos:

https://github.com/WalletConnect/web3modal/assets/26643922/1415f85a-257f-493e-b332-7f7c0a5381c3

**Comment by glitch-txs** [comment]
@nsnull0 I'm not able to repro and I cannot see the video idk why.

Which wallet are you using? I tested it with Trust (mobile) and MetaMask (extension) both work for me on page refreash

**Comment by nsnull0** [comment]
https://github.com/WalletConnect/web3modal/assets/26643922/59617759-2752-4c91-9ea5-61f558ccc304

@glitch-txs i don't know man, its just metamask and now is pretty easy to reproduce.
I can reproduce it on https://web3modal.com too
just connect with metamask wallet extention.

> I cannot see the video idk why.

you mean you can't see my videos at here? would you like me to upload to youtube? I could you send it to your email as well to check if want.. just let me know.

**Comment by glitch-txs** [comment]
@nsnull0 thanks! I was able to see the last video, I was only able to reproduce this on Edge but I see some other devs are experiencing it as well, we'll try to fix this today, as for now we know this happens when the email feature is enabled.

**Comment by glitch-txs** [comment]
We'll follow up on this issue, feel free to subscribe and comment. https://github.com/WalletConnect/web3modal/issues/2126

Thanks! 👍

**Comment by nsnull0** [comment]
wow very fast,, thank you.. i'll try soon..

**Comment by nsnull0** [comment]
alright, it's fixed at 4.1.7 finally
